### PR TITLE
Change _MMInitializeDataLoader.get_inputs() output format

### PR DESCRIPTION
**Pull request by kprokofi**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by kprokofi** [comment]
CVS-112024
Need to change dataloader format from train to validation and use test functions on mmcv side

File: otx/algorithms/classification/adapters/mmcls/nncf/builder.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def build_nncf_classifier(  # pylint: disable=too-many-locals,too-many-statement
                 dataset_builder=mmcls_build_dataset,
             ),
             config,
-            subset="train",
+            subset="val",
```
Comment (cih9088) [inline otx/algorithms/classification/adapters/mmcls/nncf/builder.py:83]:
```suggestion
            subset="train",
```
The quantisation parameter init dataset should be a train subset not a validation subset.

File: otx/algorithms/detection/adapters/mmdet/nncf/builder.py [hunk 2]
```diff
@@ -91,7 +91,7 @@ def build_nncf_detector(  # pylint: disable=too-many-locals,too-many-statements
                 dataset_builder=mmdet_build_dataset,
             ),
             config,
-            subset="train",
+            subset="val",
```
Comment (cih9088) [inline otx/algorithms/detection/adapters/mmdet/nncf/builder.py:94]:
```suggestion
            subset="train",
```

File: otx/algorithms/segmentation/adapters/mmseg/nncf/builder.py [hunk 3]
```diff
@@ -92,7 +92,7 @@ def build_nncf_segmentor(  # noqa: C901  # pylint: disable=too-many-locals,too-m
                 dataset_builder=mmseg_build_dataset,
             ),
             config,
-            subset="train",
+            subset="val",
```
Comment (cih9088) [inline otx/algorithms/segmentation/adapters/mmseg/nncf/builder.py:95]:
```suggestion
            subset="train",
```

**Review by cih9088 — CHANGES_REQUESTED** [review]

**Comment by cih9088** [comment]
@nikita-savelyevv, Could you please double-check if it is a bug or not since it was implemented in this way when we maintained a forked mm repositories like following.
https://github.com/openvinotoolkit/mmsegmentation/blob/ote/mmseg/integration/nncf/compression.py#L125-L130
https://github.com/openvinotoolkit/mmdetection/blob/ote/mmdet/integration/nncf/compression.py#L127-L132

@kprokofi, It is not in hurry but we should investigate `ignore_scopes` in each `compression_config.json` as this change would make different NNCF internal graph for some models. We might not need some of them in `ignore_scopes`. If it is not included in this PR, at least we should add `TODO` comment somewhere.

**Comment by kprokofi** [comment]
@cih9088 , I use not val, but train. It just constructs images differently (1 image per batch) for NNCF initialization. 
Moreover, I discussed with NNCF what subset is better to use for init procedure and it was suggested to me using validation subset instead of train, but I left train for now.
If we use train subset in dataloader - it will lead a problem in simple_test() evaluation function

I will add TODO for future ignore_scope investigation

File: otx/algorithms/classification/adapters/mmcls/nncf/builder.py (hunk 1 above)
Comment (kprokofi) [reply otx/algorithms/classification/adapters/mmcls/nncf/builder.py:83]: We use train images, with "val" parameter  we construct batch differently, in the way we need for NNCF initialization

Comment (cih9088) [reply otx/algorithms/classification/adapters/mmcls/nncf/builder.py:83]: ahh, sorry, my bad. it was for dataloader not dataset.

**Comment by kprokofi** [comment]
Locally, tests are passed (for cls). Waiting for CI. Some problem with connection also observed for detection tests
image

**Comment by kprokofi** [comment]
Tests are passed

**Review by cih9088 — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Pull request merged**